coral/src/app/features/topics/request/TopicPromotionRequest.tsx

```diff
@@ -0,0 +1,282 @@
+      advancedConfiguration: "{\n}",
+    },
+  });
+
+  useEffect(() => {
+    if (
+      targetEnvironment !== undefined &&
+      targetEnvironment.params !== undefined &&
+      topicDetailsForSourceEnv !== undefined
+    ) {
+      form.setValue("environment", targetEnvironment);
+      form.setValue(
+        "topicpartitions",
+        String(targetEnvironment?.params.defaultPartitions)
+      );
+      form.setValue(
+        "replicationfactor",
+        String(targetEnvironment?.params.defaultRepFactor)
+      );
+
+      if (topicDetailsForSourceEnv.topicContents?.description !== undefined) {
+        form.setValue(
+          "description",
+          topicDetailsForSourceEnv.topicContents.description
+        );
+      }
+
+      if (
+        topicDetailsForSourceEnv.topicContents?.advancedTopicConfiguration !==
+        undefined
+      ) {
+        form.setValue(
+          "advancedConfiguration",
+          JSON.stringify(
+            topicDetailsForSourceEnv.topicContents.advancedTopicConfiguration
+          )
+        );
+      }
+    }
+  }, [targetEnvironment, topicDetailsForSourceEnv]);
+
+  const {
+    mutate: promote,
+    isLoading: promoteIsLoading,
+    isError: promoteIsError,
+    error: promoteError,
+  } = useMutation(promoteTopic, {
+    onSuccess: () => {
+      navigate(-1);
+      toast({
+        message: "Topic promotion request successfully created",
+        position: "bottom-left",
+        variant: "default",
+      });
+    },
+  });
+
+  const onPromoteSubmit: SubmitHandler<Schema> = (data) => promote(data);
+
+  function cancelRequest() {
+    form.reset();
+    navigate(-1);
+  }
+
+  return (
+    <>
+      {cancelDialogVisible && (
+        <Dialog
+          title={`Cancel topic promotion request?`}
+          primaryAction={{
+            text: `Cancel request`,
+            onClick: () => cancelRequest(),
+          }}
+          secondaryAction={{
+            text: `Continue with request`,
+            onClick: () => setCancelDialogVisible(false),
+          }}
+          type={"warning"}
+        >
+          Do you want to cancel this request? The data added will be lost.
+        </Dialog>
+      )}
+      <Box>
+        <Form
+          {...form}
+          ariaLabel={"Request topic promotion"}
+          onSubmit={onPromoteSubmit}
+        >
+          <Box width={"full"}>
+            {targetEnvironment !== undefined ? (
+              <NativeSelect
+                name="environment"
+                labelText={"Environment"}
+                required
+                readOnly
+              >
+                <Option
+                  key={targetEnvironment.name}
+                  value={targetEnvironment.name}
+                >
+                  {targetEnvironment.name}
+                </Option>
+              </NativeSelect>
+            ) : (
+              <NativeSelect.Skeleton />
+            )}
+          </Box>
+          <Box>
+            <Box paddingY={"l1"}>
```

Comment:
Which component? And what kind of implementation are you thinking of? But anyway, this comment brought the fact that there are a lot of unnecessary boxes in this component :o [1f90ea3](https://github.com/aiven/klaw/pull/1420/commits/1f90ea31f836e4d5fe7d3715995c1bf3e6db5a9e)

Reply:
On one hand, yes it would be nice to be able to add the spacing on the divider. On the other, this feels like a deliberate decision on the part of the `aquarium` team? But we probably could bring it up ^^